sympy/combinatorics/pc_groups.py

```diff
@@ -75,63 +76,55 @@ def __init__(self, pc_relators, relative_order, group):
         self.relative_order = relative_order
         self.group = group
 
-    def minimal_uncollected_subwords(self, word):
+    def minimal_uncollected_subword(self, word):
         """
         Returns the minimal uncollected subwords.
 
         Examples
         ========
         >>> from sympy.combinatorics.pc_groups import Collector
         >>> from sympy.combinatorics.free_groups import free_group
+        >>> from sympy.core import S
 
         Example 8.7 Pg. 281 from [1]
         >>> F, x1, x2 = free_group("x1, x2")
         >>> pc_relators = {x1**2 : (), x1*x2*x1**-1 : x2**-1, x1**-1*x2*x1 : x2**-1}
-        >>> relative_order = {x1: 2, x2: 3}
+        >>> relative_order = {x1: 2, x2: S.Infinity}
         >>> word = x2**2*x1**7
         >>> group = word.group
         >>> collector = Collector(pc_relators, relative_order, group)
-        >>> collector.minimal_uncollected_subwords(word)
-        {((x1, 7),): 2, ((x2, 2), (x1, 1)): 0}
+        >>> collector.minimal_uncollected_subword(word)
+        ((x2, 2), (x1, 1))
 
         """
-
         # To handle the case word = <identity>
         if not word:
-            return {}
+            return None
+
         group = self.group
         index = {s: i+1 for i, s in enumerate(group.symbols)}
         array = word.array_form
         re = self.relative_order
-        uncollected_subwords = {}
 
         for i in range(len(array)-1):
             s1, e1 = array[i]
             s2, e2 = array[i+1]
             s = ((s1, 1), )
             s = group.dtype(s)
-            if e1 > re[s]-1:
-                # case-2: v = x[i]**a
-                uncollected_subwords[((s1, e1), )] = 2
-                continue
+            if (e1 < 1 or e1 > re[s]-1) and re[s] != S.Infinity:
```

Comment:
Is it possible that `e1 == 0`? If not, then I would write `e1 < 0`. So this could be

    if s in re and (e1 < 0 or e1 >= re[s]):

Reply:
> Is it possible that `e1 == 0`?

No!